### ISSUE TITLE
graphql pageInfo.hasNextPage did not work properly after upgrade packages to latest version

### ISSUE THREAD

**Issue by hgb123** [gatsbyjs/gatsby#32485]
### Preliminary Checks

- [X] This issue is not a duplicate. Before opening a new issue, please search existing issues: https://github.com/gatsbyjs/gatsby/issues
- [X] This issue is not a question, feature request, RFC, or anything other than a bug report directly related to Gatsby. Please post those things in GitHub Discussions: https://github.com/gatsbyjs/gatsby/discussions

### Description

I upgraded all package version to the latest (information below) and the pagination feature broke due to the unexpected value on `pageInfo.hasNextValue`

Page 1, Next button is enabled ✅
![image](https://user-images.githubusercontent.com/18468577/126637302-0c3cd830-a341-40f2-a55e-e6048583de73.png)

Page 2, Next button is disabled ❌
![image](https://user-images.githubusercontent.com/18468577/126637250-002c82cf-70ad-47a1-8af2-35a6411bb748.png)

Page 3, Next button is disabled ❌
![image](https://user-images.githubusercontent.com/18468577/126637275-b13ea057-effc-4c2d-a30d-ef6f88d3c64a.png)

### Reproduction Link

https://github.com/hgb123/gatsby-issue-reproduce

### Steps to Reproduce

1. Install packages `yarn`
2. Run local dev `yarn start`
3. Open `localhost:8000`
4. Click `Next` button

### Expected Result

`pageInfo.hasNextPage` should be `true` on page 1-3/4

### Actual Result

`pageInfo.hasNextPage` was `false` on page 2/4, 3/4

### Environment

```shell
System:
    OS: Windows 10 10.0.19042
    CPU: (8) x64 Intel(R) Core(TM) i7-4810MQ CPU @ 2.80GHz
  Binaries:
    Node: 14.17.0 - D:\Harrogath\nodejs\node.EXE
    Yarn: 1.22.10 - ~\AppData\Roaming\npm\yarn.CMD
    npm: 7.18.1 - D:\Harrogath\nodejs\npm.CMD
  Languages:
    Python: 3.8.5
  Browsers:
    Chrome: 91.0.4472.164
    Edge: Spartan (44.19041.1023.0), Chromium (91.0.864.71)
  npmPackages:
    gatsby: 3.10.1 => 3.10.1 
    gatsby-source-filesystem: 3.10.0 => 3.10.0 
    gatsby-transformer-remark: ^4.7.0 => 4.7.0 
  npmGlobalPackages:
    gatsby-cli: 3.4.1
```

**Comment by vladar** [comment]
Thank you for the report and the great repro! Should be hopefully fixed in #32496

**Comment by LekoArts** [comment]
Fixed in `gatsby@3.10.2`
